pkg/agent/statusupdate.go

```diff
@@ -68,18 +86,41 @@ func (su *periodicStatusUpdate) Execute() error {
 	return nil
 }
 
-// StartPeriodicStatusUpdate - starts a job that runs the periodic status updates
-func StartPeriodicStatusUpdate() {
+// StartAgentStatusUpdate - starts 2 separate jobs that runs the periodic status updates and immediate status updates
+func StartAgentStatusUpdate() {
+	startPeriodicStatusUpdate()
+	startImmediateStatusUpdate()
+}
+
+// startPeriodicStatusUpdate - start periodic status updates based on report activity frequency config
+func startPeriodicStatusUpdate() {
 	interval := agent.cfg.GetReportActivityFrequency()
-	statusUpdate = &periodicStatusUpdate{}
-	_, err := jobs.RegisterDetachedIntervalJob(statusUpdate, interval)
+	periodicStatusUpdate = &agentStatusUpdate{
+		typeOfStatusUpdate: periodic,
+	}
+	_, err := jobs.RegisterDetachedIntervalJob(periodicStatusUpdate, interval)
 
 	if err != nil {
 		log.Error(errors.Wrap(errors.ErrStartingPeriodicStatusUpdate, err.Error()))
 	}
 }
 
-func (su *periodicStatusUpdate) getCombinedStatus() string {
+// startImmediateStatusUpdate - start job that will 'immediately' update status.  NOTE : By 'immediately', this means currently 10 seconds.
+// The time interval for this job is hard coded.
+func startImmediateStatusUpdate() {
+	interval := 10 * time.Second
```

Comment:
Hard coded to 10 seconds.  This seems reasonable to me instead of 1s...  I could even go 15 or 20 seconds...

Reply:
I am alright with this as well, its just a check every 10 secs